step/step.go

```diff
@@ -94,18 +95,18 @@ func (e EnvVarSharer) parseEnvVars(s string) ([]EnvVar, error) {
 
 	lines := strings.Split(s, "\n")
 	for _, line := range lines {
-		split := strings.Split(line, "=")
-		if len(split) > 2 || len(split) == 0 {
-			return nil, fmt.Errorf("env var should be in a format: KEY=value or KEY: %s", line)
+		key, value, found := strings.Cut(line, "=")
+		// empty line
+		if key == "" {
+			if found {
+				return nil, fmt.Errorf("env var should be in a format: KEY=value or KEY: %s", line)
+			} else {
+				// empty line is ignored
+				continue
+			}
```

Comment:
I needed some time to understand the cases here. Then I wondered if we could simply this by extracting the empty line case and do something like this:
```
func (e EnvVarSharer) parseEnvVars(s string) ([]EnvVar, error) {
	var envVars []EnvVar

	lines := strings.Split(s, "\n")
	for _, line := range lines {
		if strings.TrimSpace(line) == "" {
			// empty line is ignored
			continue
		}

		key, value, _ := strings.Cut(line, "=")
		if key == "" {
			return nil, fmt.Errorf("env var should be in a format: KEY=value or KEY: %s", line)
		}
		if value == "" {
			value = e.envRepository.Get(key)
		}

		envVars = append(envVars, EnvVar{
			Key:   key,
			Value: value,
		})
	}

	return envVars, nil
}
```
So we could start by checking for an empty line and skip the string cutting. This way we also do not need to check the `found` bool and just rely on the key itself.

Reply:
done